syntax = "proto3";

package simpleStream;

service SimpleStream {
	rpc ServerUpload(stream SendRequest) returns(SendResponse) {}
	rpc ClientDownload(RecvRequest) returns(stream RecvResponse) {}
}

message StreamHeader {
    string type = 1;
    uint64 size = 2;
}

message StreamPayload {
    bytes data = 1;
}

message SendRequest {
    oneof contents {
        StreamHeader header = 1;
        StreamPayload payload  = 2;
    }
}

message SendResponse {
    string type = 1;
    uint64 size = 2;
}

message RecvRequest {
    string type = 1;
    uint64 size = 2;
}

message RecvResponse {
	oneof contents {
        StreamHeader header = 1;
        StreamPayload payload  = 2;
    }
}